ddev delete images results in runtime error: index out of range [0] with length 0

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
======= Creating dummy project named  tryddevproject-2943 in ../tryddevproject-2943 =========
OS Information: Darwin MacBook-Pro.local 21.6.0 Darwin Kernel Version 21.6.0: Mon May 22 20:41:03 PDT 2023; root:xnu-8020.240.18.702.2~1/RELEASE_X86_64 x86_64
ProductName:    macOS
ProductVersion: 12.6.7
BuildVersion:   21G708
User information: uid=501(macos_username) gid=20(staff) groups=20(staff),12(everyone),61(localaccounts),79(_appserverusr),80(admin),81(_appserveradm),98(_lpadmin),398(com.apple.access_screensharing),701(com.apple.sharepoint.group.1),33(_appstore),100(_lpoperator),204(_developer),250(_analyticsusers),395(com.apple.access_ftp),399(com.apple.access_ssh),400(com.apple.access_remote_ae)
DDEV version:  ITEM             VALUE                                   
 DDEV version     v1.21.6                                 
 architecture     amd64                                   
 db               drud/ddev-dbserver-mariadb-10.4:v1.21.5 
 dba              phpmyadmin:5                            
 ddev-ssh-agent   drud/ddev-ssh-agent:v1.21.5             
 docker           24.0.2                                  
 docker-compose   v2.15.1                                 
 docker-platform  docker-desktop                          
 mutagen          0.16.0                                  
 os               darwin                                  
 router           drud/ddev-router:v1.21.5                
 web              drud/ddev-webserver:v1.21.5             
PROXY settings: HTTP_PROXY='' HTTPS_PROXY='' http_proxy='' NO_PROXY=''
======= DDEV global info =========
Global configuration:
instrumentation-opt-in=false
omit-containers=[]
mutagen-enabled=false
nfs-mount-enabled=true
router-bind-all-interfaces=false
internet-detection-timeout-ms=3000
disable-http2=false
use-letsencrypt=false
letsencrypt-email=
table-style=default
simple-formatting=false
auto-restart-containers=false
use-hardened-images=false
fail-on-hook-fail=false
required-docker-compose-version=
use-docker-compose-from-path=false
project-tld=
xdebug-ide-location=
no-bind-mounts=false
use-traefik=false
wsl2-no-windows-hosts-mgt=false

======= DOCKER info =========
docker location: lrwxr-xr-x  1 root  wheel  54 Feb  4  2022 /usr/local/bin/docker -> /Applications/Docker.app/Contents/Resources/bin/docker
Docker Desktop Version: Docker Desktop for Mac 4.20.0 build 109717
docker version: 
Client:
 Cloud integration: v1.0.33
 Version:           24.0.2
 API version:       1.43
 Go version:        go1.20.4
 Git commit:        cb74dfc
 Built:             Thu May 25 21:51:16 2023
 OS/Arch:           darwin/amd64
 Context:           default

Server: Docker Desktop 4.20.0 (109717)
 Engine:
  Version:          24.0.2
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.4
  Git commit:       659604f
  Built:            Thu May 25 21:52:17 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
DOCKER_DEFAULT_PLATFORM=notset
======= Mutagen Info =========
Mutagen is installed in ddev, version=0.16.0
Started Mutagen daemon in background (terminate with "mutagen daemon stop")     
--------------------------------------------------------------------------------
Name: profeminaorg
Identifier: sync_6F2FDbgKDpIDrzjrJO8LWIwUxm1z8b7J7wsPs3IlAPj
Labels:
        com.ddev.volume-signature: var-run-1667572002
Configuration:
        Synchronization mode: Two Way Resolved
        Maximum allowed entry count: Default (2⁶⁴−1)
        Maximum staging file size: Default (18 EB)
        Symbolic link mode: POSIX Raw
        Ignore VCS mode: Default (Propagate)
        Ignores:
                /.git
                /.tarballs
                /.ddev/db_snapshots
                /.ddev/.importdb*
                .DS_Store
                .idea
                /web/sites/default/files
        Permissions mode: Default (Portable)
Alpha:
        URL: /Users/macos_username/Local projects/profemina.org
        Configuration:
                Watch mode: Default (Portable)
                Watch polling interval: Default (10 seconds)
                Probe mode: Default (Probe)
                Scan mode: Default (Accelerated)
                Stage mode: Neighboring
                File mode: Default (0600)
                Directory mode: Default (0700)
                Default file/directory owner: Default
                Default file/directory group: Default
        Connected: No
Beta:
        URL: docker://ddev-profemina.org-web/var/www/html
                DOCKER_HOST=unix:///var/run/docker.sock
        Configuration:
                Watch mode: Default (Portable)
                Watch polling interval: Default (10 seconds)
                Probe mode: Default (Probe)
                Scan mode: Default (Accelerated)
                Stage mode: Neighboring
                File mode: Default (0600)
                Directory mode: Default (0700)
                Default file/directory owner: Default
                Default file/directory group: Default
        Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
======= Docker Info =========
Docker platform: docker-desktop 
Using docker context: default (unix:///Users/macos_username/.docker/run/docker.sock) 
docker-compose: v2.15.1 
Using DOCKER_HOST=unix:///Users/macos_username/.docker/run/docker.sock 
Docker version: 24.0.2 
Able to run simple container that mounts a volume. 
Able to use internet inside container. 
Docker disk space:
Filesystem                Size      Used Available Use% Mounted on
overlay                  78.4G     62.8G     11.6G  84% /

Stopped mutagen daemon 
The ddev-ssh-agent container has been removed. When you start it again you will have to use 'ddev auth ssh' to provide key authentication again. 
Existing docker containers: 
CONTAINER ID   IMAGE            COMMAND                  CREATED         STATUS                      PORTS              NAMES
64897cc37d99   09e5271db9c9     "sh -c '( test -f /v…"   3 months ago    Exited (0) 3 months ago                        bold_taussig
329dfb752867   09e5271db9c9     "sh -c 'chown -R 501…"   3 months ago    Exited (0) 3 months ago                        beautiful_blackburn
0bcd74a0a367   734e7aa30976     "/bin/sh"                4 months ago    Created                                        phpstorm_helpers_PS-223.8617.59
e93d305a094e   09e5271db9c9     "sh -c '( test -f /v…"   4 months ago    Exited (0) 4 months ago                        zealous_haibt
d65f32f8e2ae   09e5271db9c9     "sh -c 'chown -R 501…"   4 months ago    Exited (0) 4 months ago                        dazzling_ishizaka
d50112ba18b3   09e5271db9c9     "sh -c 'chown -R 501…"   5 months ago    Exited (0) 5 months ago                        inspiring_boyd
8849d001785d   c94c708da410     "/bin/sh"                5 months ago    Created                                        phpstorm_helpers_PS-223.8214.64
6c18a6cfb167   02c6ce4f7b23     "/bin/sh"                6 months ago    Created                                        phpstorm_helpers_PS-223.7571.212
a2171e3634b3   2696934c2d22     "/bin/sh"                6 months ago    Created                                        phpstorm_helpers_PS-222.4459.28
86425e138809   e0417e88c0bd     "sh -c 'chown -R 501…"   6 months ago    Exited (0) 6 months ago                        vibrant_bardeen
17d40f1c846b   e0417e88c0bd     "sh -c 'chown -R 501…"   6 months ago    Exited (0) 6 months ago                        frosty_clarke
e89dd17f85dc   e0417e88c0bd     "sh -c 'chown -R 501…"   7 months ago    Exited (0) 7 months ago                        funny_tesla
b03f9c009905   1d4902d9c203     "/bin/sh"                7 months ago    Created                                        phpstorm_helpers_PS-222.4345.15
e32eed880aea   0208efb41aa5     "sh -c '( test -f /v…"   8 months ago    Exited (0) 8 months ago                        silly_stonebraker
7b2afe6f8307   0208efb41aa5     "sh -c 'chown -R 501…"   8 months ago    Exited (0) 8 months ago                        trusting_lumiere
454ed38b0199   0208efb41aa5     "sh -c 'mkdir -p /mn…"   8 months ago    Exited (255) 8 months ago   80/tcp, 8025/tcp   CopyIntoVolume_zdandbhfjvfs
ef6efbaa1931   busybox:stable   "sh -c 'rm -f /mnt/d…"   9 months ago    Created                                        xenodochial_ishizaka
0407dce1964d   busybox:stable   "sh -c 'rm -f /mnt/d…"   9 months ago    Created                                        optimistic_curran
fd43ee17a665   0208efb41aa5     "sh -c '( test -f /v…"   9 months ago    Exited (0) 9 months ago                        quizzical_mccarthy
ff75527cb26b   0208efb41aa5     "sh -c '( test -f /v…"   9 months ago    Exited (0) 9 months ago                        mystifying_bartik
dde6578bdfce   36bfeda8cd6e     "sh -c 'chown -R 501…"   10 months ago   Exited (0) 10 months ago                       mystifying_pike
Creating a new ddev project config in the current directory (/Users/macos_username/Local projects/tryddevproject-2943) 
Once completed, your configuration will be written to /Users/macos_username/Local projects/tryddevproject-2943/.ddev/config.yaml
 
Configuring unrecognized codebase as project type 'php' at /Users/macos_username/Local projects/tryddevproject-2943/web 
Configuration complete. You may now run 'ddev start'. 
Network ddev_default created 
Starting tryddevproject-2943... 
Container ddev-ssh-agent  Started 
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys. 
Network ddev-tryddevproject-2943_default  Created 
Volume "ddev-tryddevproject-2943_nfsmount"  Created 
Container ddev-tryddevproject-2943-dba  Started 
Container ddev-tryddevproject-2943-db  Started 
Container ddev-tryddevproject-2943-web  Started 
Container ddev-router  Started 
Successfully started tryddevproject-2943 
Project can be reached at https://tryddevproject-2943.ddev.site https://127.0.0.1:62212 
======== Curl of site from inside container:
HTTP/1.1 200 OK
Server: nginx
Date: Tue, 13 Jun 2023 13:31:41 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== curl -I of http://tryddevproject-2943.ddev.site from outside:
HTTP/1.1 200 OK
Server: nginx/1.20.1
Date: Tue, 13 Jun 2023 13:31:41 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== full curl of http://tryddevproject-2943.ddev.site from outside:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-2943'
======== Project ownership on host:
drwxr-xr-x  4 macos_username  staff  128 Jun 13 15:30 ../tryddevproject-2943
======== Project ownership in container:
drwxr-xr-x 4 macos_username dialout 128 Jun 13 13:30 /var/www/html
======== In-container filesystem:
Filesystem                                                                 Type 1K-blocks      Used Available Use% Mounted on
:/System/Volumes/Data/Users/macos_username/Local projects/tryddevproject-2943 nfs  488245312 451048096  37197216  93% /var/www/html
======== curl again of tryddevproject-2943 from host:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-2943'
Thanks for running the diagnostic. It was successful.
Please provide the output of this script in a new gist at gist.github.com
Running ddev launch in 5 seconds
If you're brave and you have jq you can delete all tryddevproject instances with this one-liner:
    ddev delete -Oy $(ddev list -j |jq -r .raw[].name | grep tryddevproject)
In the future ddev debug test will also provide this option.

Please delete this project after debugging with 'ddev delete -Oy tryddevproject-2943'

```
</details>         


### Expected Behavior

`ddev delete images -y` removes outdated images.

### Actual Behavior

`ddev delete images -y`

Now returns an error;

```
Powering off ddev to avoid conflicts 
Stopped mutagen daemon 
The ddev-ssh-agent container has been removed. When you start it again you will have to use 'ddev auth ssh' to provide key authentication again. 
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/ddev/ddev/cmd/ddev/cmd.deleteDdevImages.func1(0xc00005a900?, 0xc000366ae0?)
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/delete-images.go:93 +0x9b
sort.order2_func(...)
        /opt/hostedtoolcache/go/1.20.1/x64/src/sort/zsortfunc.go:299
sort.median_func({0xc0000f7c30?, 0xc000366ae0?}, 0x17, 0x18, 0x19, 0xc0000f7970)
        /opt/hostedtoolcache/go/1.20.1/x64/src/sort/zsortfunc.go:308 +0x4a
sort.medianAdjacent_func(...)
        /opt/hostedtoolcache/go/1.20.1/x64/src/sort/zsortfunc.go:316
sort.choosePivot_func({0xc0000f7c30?, 0xc000366ae0?}, 0x0, 0x1cbf760?)
        /opt/hostedtoolcache/go/1.20.1/x64/src/sort/zsortfunc.go:279 +0xa8
sort.pdqsort_func({0xc0000f7c30?, 0xc000366ae0?}, 0x1e09a68?, 0x18?, 0xc00005a900?)
        /opt/hostedtoolcache/go/1.20.1/x64/src/sort/zsortfunc.go:89 +0xdb
sort.Slice({0x1636c40?, 0xc000010120?}, 0x8d?)
        /opt/hostedtoolcache/go/1.20.1/x64/src/sort/slice.go:26 +0xfa
github.com/ddev/ddev/cmd/ddev/cmd.deleteDdevImages(0x0)
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/delete-images.go:89 +0xe7
github.com/ddev/ddev/cmd/ddev/cmd.glob..func23(0x1c79e20?, {0xc000245970?, 0x1?, 0x1?})
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/delete-images.go:43 +0x9b
github.com/spf13/cobra.(*Command).execute(0x1c79e20, {0xc000245960, 0x1, 0x1})
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x1c7dca0)
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:974 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:902
github.com/ddev/ddev/cmd/ddev/cmd.Execute()
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/root.go:143 +0x2f
main.main()
        /home/runner/work/ddev/ddev/cmd/ddev/main.go:20 +0x9b
```

This has been the case for a while

### Steps To Reproduce

_No response_

### Anything else?

_No response_

--- a/cmd/ddev/cmd/delete-images.go
+++ b/cmd/ddev/cmd/delete-images.go
@@ -87,7 +87,10 @@ func deleteDdevImages(deleteAll bool) error {
 	// Sort so that images that have -built on the end
 	// come up before their parent images that don't
 	sort.Slice(images, func(i, j int) bool {
-		if images[i].RepoTags == nil || images[j].RepoTags == nil {
+		if images[i].RepoTags == nil || len(images[i].RepoTags) == 0 {
+			return false
+		}
+		if images[j].RepoTags == nil || len(images[j].RepoTags) == 0 {
 			return false
 		}
 		return images[i].RepoTags[0] > images[j].RepoTags[0]

--- a/cmd/ddev/cmd/delete-images.go
+++ b/cmd/ddev/cmd/delete-images.go
@@ -111,6 +111,12 @@ func deleteDdevImages(deleteAll bool) error {
 	// and discrete names of images
 	for _, image := range images {
 		for _, tag := range image.RepoTags {
+			// If anything has prefix "drud/" then delete it
+			if strings.HasPrefix(tag, "drud/") {
+				if err = dockerutil.RemoveImage(tag); err != nil {
+					return err
+				}
+			}
 			// If a webimage, but doesn't match our webimage, delete it
 			if strings.HasPrefix(tag, versionconstants.WebImg) && !strings.HasPrefix(tag, webimg) && !strings.HasPrefix(tag, webimg+"-built") {
 				if err = dockerutil.RemoveImage(tag); err != nil {

